uint16 number
uint16 predict_step
float64[] x
float64[] y